src/db/postgres/lrsql/postgres/sql/ddl.sql

```diff
@@ -358,3 +358,9 @@ ALTER TABLE activity ALTER COLUMN payload SET DATA TYPE JSONB;
 ALTER TABLE xapi_statement ALTER COLUMN payload SET DATA TYPE JSON;
 ALTER TABLE actor ALTER COLUMN payload SET DATA TYPE JSON;
 ALTER TABLE activity ALTER COLUMN payload SET DATA TYPE JSON;
+
+-- :name add-statement-to-actor-cascading-delete!
+-- :command :execute
+-- :doc Adds a cascading delete to delete st2actor entries when corresponding statements are deleted
+ALTER TABLE statement_to_actor DROP CONSTRAINT statement_fk;
+ALTER TABLE statement_to_actor ADD CONSTRAINT statement_fk FOREIGN KEY (statement_id) REFERENCES xapi_statement(statement_id) ON DELETE CASCADE;
```

Comment:
I am not really seeing the idempotence in here. You need EITHER to check in another query if this is already done or do an if-not-exists kinda deal in this query. I see you dropping it but keep in mind it might take _minutes_ to populate an index on a sufficiently large db. This would mean that every startup takes minutes to run.

Reply:
`add-statement-to-actor-cascading-delete!` is now gated on `check-statement-to-actor-cascading-delete` in `update-all!` of `postgres/record.clj`.  There will still be a re-indexing, but only once.